src/Workspaces/Core/Portable/Options/GlobalOptionService.cs

```diff
@@ -91,12 +95,28 @@ static ImmutableHashSet<IOption> ComputeSerializableOptionsFromProviders(Immutab
             }
         }
 
+        private ImmutableArray<IOptionPersister> GetOptionPersisters()
+        {
+            if (_lazyOptionSerializers is { IsDefault: false } optionSerializers)
```

Comment:
How is this better than `if (!_lazyOptionSerializers.IsDefault)`?

Reply:
it also puts the result in a new local variable

Reply:
And that is useful because _lazyOptionServializers might be modified on a different thread?

Reply:
Yes, it is initialized a few lines below using `InterlockedInitialize`